### PR TITLE
*: combine etcdctl and etcdctlv3

### LINKED ISSUE #4877: proposal: combine etcdctl and etcdctlv3 into one binary
Now we have two binaries for v2/v3 API. We want to combine them into one binary to make it easy to deliver released binary to users. But we also do not want to confuse users with some magics to guess with version of API the user intend to use. Sometimes it is also impossible to infer from user input without being explicitly (for example, getting a key when the server-side has both v2 and v3 API enabled).

The proposed solution is simple (implementation and usage).

To use v3 API, user explicitly set env `ETCDCTL_V3`
```
ETCDCTL_V3=true ./etcdctl put foo bar
```

To use v3 API, user set nothing.

```
./etcdctl set foo bar
```

To implement this, we just need to add a few lines into main func

```go
isV3 := os.Getenv("ETCDCTL_V3")
os.Setenv("ETCDCTL_V3", "") // clear env after use. so it will not affect any command parsing.
if useV3(isV3) {
    v3()
    return
}
v2()
```

### PR DESCRIPTION
**Pull request by xiang90**
Fix #4877

### CODE REVIEW

File: etcdctl/main.go [hunk 1]
```diff
@@ -12,59 +12,34 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// etcdctl is a command line application that controls etcd.
```
Comment (heyitsanthony) [inline etcdctl/main.go]: I think this godoc is still OK?

File: etcdctl/ctlv3/ctl.go [hunk 2]
```diff
@@ -12,8 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// etcdctlv3 is a command line application that utilizes v3 API.
-package main
+// ctlv3 is a command line application that utilizes v3 API.
```
Comment (heyitsanthony) [inline etcdctl/ctlv3/ctl.go]: I don't think this is accurate if there's no ctlv3 binary

File: V3DemoProcfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Use goreman to run `go get github.com/mattn/goreman`
-# etcd1 is the default client server for etcdctlv3 commands
+# etcd1 is the default client server for etcdctl commands
```
Comment (gyuho) [inline V3DemoProcfile]: I think default endpoints are all of these 3. https://github.com/coreos/etcd/blob/master/etcdctlv3/main.go#L48

Maybe just remove this line?

**Comment by gyuho** [comment]
lgtm.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**